### PR TITLE
Change Signature - Fix 'remove parameter' removing arguments with '@' 

### LINKED ISSUE #66547: Refactor "Remove parameter" mistakenly removes arguments with '@'
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Refactor-Remove-parameter-mistakenly-r/10259275)._

---
The refactor "Remove parameter" misbehaves by removing arguments at call-sites that it shouldn't. In particular arguments whose parameter names start with `@` and have been explicitly specified.

I've reproduced it a minimal example shown in the screenshot.
Simply create a new C# console app and insert the following in its only C# file:

```csharp
void m(object? param, bool @new = true)
{

}

m(null);
m(null, false);
m(null, @new: false);

```
Execute the refactor "Remove parameter" on and remove `param`, and observe that the argument `@new: false` on the last line is mistakenly removed. I expected that argument not to be removed.

### Version

Reproduced in Visual Studio version 17.4.4.

---
#### Feedback Bot on 1/23/2023, 06:38 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

### PR DESCRIPTION
**Pull request by akhera99**
Fixes #66547

### CODE REVIEW

File: src/Features/CSharp/Portable/ChangeSignature/UnifiedArgumentSyntax.cs [hunk 1]
```diff
@@ -47,7 +47,7 @@ public IUnifiedArgumentSyntax WithNameColon(SyntaxNode nameColonSyntax)
         }
 
         public string GetName()
-            => NameColon == null ? string.Empty : ((NameColonSyntax)NameColon).Name.Identifier.ToString();
+            => NameColon == null ? string.Empty : ((NameColonSyntax)NameColon).Name.Identifier.ValueText;
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/ChangeSignature/UnifiedArgumentSyntax.cs:50]: do we have the same issue for VB?  can you add a test?  note: in vb, the form is `[new]`.

**Review by CyrusNajmabadi — APPROVED** [review]
LGTM if vb is fine.

File: src/Features/CSharp/Portable/ChangeSignature/UnifiedArgumentSyntax.cs (hunk 1 above)
Comment (akhera99) [reply src/Features/CSharp/Portable/ChangeSignature/UnifiedArgumentSyntax.cs:50]: issue doesn't repro in VB -- added a test just to be safe

File: src/EditorFeatures/VisualBasicTest/ChangeSignature/RemoveParametersTests.vb [hunk 2]
```diff
@@ -151,5 +151,33 @@ End Class]]></Text>.NormalizedValue()
...
+        <WorkItem(66547, "https://github.com/dotnet/roslyn/issues/66547")>
+        Public Async Function RemoveParameters_SpecialSymbolNamedParameter() As Task
+
+            Dim markup = <Text><![CDATA[
+Class C
+    Sub $$M(param As Object, Optional [new] As Boolean = False)
+    End Sub
+
+    Sub M2()
+        M(Nothing, new:=True)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/VisualBasicTest/ChangeSignature/RemoveParametersTests.vb:165]: don't you need this argument to use `[new]`?

**Pull request merged**